ibc-testkit/src/hosts/tendermint.rs

```diff
@@ -0,0 +1,218 @@
+use core::str::FromStr;
+
+use ibc::clients::tendermint::client_state::ClientState;
+use ibc::clients::tendermint::consensus_state::ConsensusState;
+use ibc::clients::tendermint::types::proto::v1::Header as RawHeader;
+use ibc::clients::tendermint::types::{Header, TENDERMINT_HEADER_TYPE_URL};
+use ibc::core::client::types::Height;
+use ibc::core::host::types::identifiers::ChainId;
+use ibc::core::primitives::prelude::*;
+use ibc::core::primitives::Timestamp;
+use ibc::primitives::proto::Any;
+use ibc::primitives::ToVec;
+use tendermint::block::Header as TmHeader;
+use tendermint::validator::Set as ValidatorSet;
+use tendermint_testgen::light_block::TmLightBlock;
+use tendermint_testgen::{
+    Generator, Header as TestgenHeader, LightBlock as TestgenLightBlock,
+    Validator as TestgenValidator,
+};
+
+use super::{TestBlock, TestHeader, TestHost};
+use crate::fixtures::clients::tendermint::ClientStateConfig;
+use crate::testapp::ibc::core::types::MockClientConfig;
+
+#[derive(Debug)]
+pub struct Host(ChainId);
+
+impl TestHost for Host {
+    type Block = TendermintBlock;
+    type BlockParams = BlockParams;
+    type LightClientParams = MockClientConfig;
+    type ClientState = ClientState;
+
+    fn with_chain_id(chain_id: ChainId) -> Self {
+        Self(chain_id)
+    }
+
+    fn chain_id(&self) -> &ChainId {
+        &self.0
+    }
+
+    fn generate_block(
+        &self,
+        height: u64,
+        timestamp: Timestamp,
+        params: &Self::BlockParams,
+    ) -> Self::Block {
+        TendermintBlock(
+            TestgenLightBlock::new_default_with_header(
+                TestgenHeader::new(&params.validators)
+                    .height(height)
+                    .chain_id(self.chain_id().as_str())
+                    .next_validators(&params.next_validators)
+                    .time(timestamp.into_tm_time().expect("Never fails")),
+            )
+            .validators(&params.validators)
+            .next_validators(&params.next_validators)
+            .generate()
+            .expect("Never fails"),
+        )
+    }
+
+    fn generate_client_state(
+        &self,
+        latest_block: &Self::Block,
```

Comment:
Nice! Note `MockHost` and `TendermintHost` don't contain the block data. They are just `string Host(String)`. Rather `MockContext<H>` wraps around `TestHost` and passes the block data.

But your suggestion totally makes sense if I implement `TestHost` on `MockContext` directly. Let's do that in the next PR when I implement the high-level contexts.

Reply:
Alright, that makes sense. Can we have a concise list of TODOs? So that we avoid leaving any unfinished work before merging to the `main`. Appreciate that.